irispy/iris_tools.py

```diff
@@ -802,3 +803,25 @@ def _apply_or_undo_exposure_time_correction(sequence, correction_function):
         else:
             converted_data_list.append(cube)
     return converted_data_list
+
+def calculate_dust_mask(data_array):
```

Comment:
I think this should have the option to supply a mask array and then return an altered copy of that mask with the dust pixels marked.  If a mask array isn't supplied, one can be generated.  So in practice this would be like:
``` python
def calculate_dust_mask(data_array, mask_array=None):
    if mask_array is None:
        mask_array = np.zeros_like(data_array, dtype=bool)
```

Reply:
You want to do that if the user just want the dust mask and not the previous mask ?